src/sync-external-store--indexable/store.test.ts

```diff
@@ -0,0 +1,135 @@
+import { beforeEach, afterEach, describe, it, jest } from "@jest/globals";
+import { IndexableSyncExternalStore } from "./store.js";
+import { IndexableStore } from "./types.js";
+
+describe("IndexableSyncExternalStore", () => {
+  describe("array data", () => {
+    type Item = { id: number; value: string };
+    const mockAction = jest.fn(async (value: Item) => value);
+    const mockStoreInitialState: IndexableStore<Item> = {
+      data: [
+        { id: 2112, value: "hello world" },
+        { id: 13, value: "hola mundo" },
+      ],
+      error: null,
+      state: "unsent",
+    };
+
+    class MockStore extends IndexableSyncExternalStore<IndexableStore<Item>> {
+      constructor() {
+        super(mockStoreInitialState);
+      }
+
+      private getIndex(item: Item) {
+        const snapshot = this.getSnapshot();
+        if (!Array.isArray(snapshot.data)) {
+          throw new Error("IndexableSyncExternalStore: data is not an array");
+        }
+        const index = snapshot.data.findIndex((data) => data.id === item.id);
```

Comment:
Mmm not quite... you should snapshot.data should be an array (which is an object). You should be able to just do `snapshot.data[data.id] = data` to set the data. This is a map lookup. Indexible means you are looking things based on a particular index-able value.